core/util/sanctionsearch.py

```diff
@@ -0,0 +1,174 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import json
+from bs4 import BeautifulSoup as bs
+
+class main:
+
+		def __init__(self, name, id=None, typ='individual', limit=15):
+			""" Sanctionsearch.ofac.treas.gov search engine
+					name     : name to search
+			"""
+			self.framework = main.framework
+			self._name = name
+			self._type = typ
+			self._listid = 'ALL'
+			self._max = limit
+			self._id = id
+			self._json = ''
+			self.sanctionsearch = 'https://sanctionssearch.ofac.treas.gov/'
+			self._rows = []
+
+			if self._name is not None:
+				self._data = []
+			else:
+				self._data = {}
+
+
+		def name_crawl(self):
+			self._name = self.framework.urlib(self._name).quote
+			self.framework.verbose('Searching sanctionsearch...')
+			headers = {
+				'Content-Type': 'application/x-www-form-urlencoded',
+				}
+
+			try:
+				# First request to get viewstate from input tag
+				req = self.framework.request(url=self.sanctionsearch)
+				soup = bs(req.text,'html.parser')
+				viewstate = soup.find('input',
+					{'type':'hidden',
+					'name':'__VIEWSTATE',
+					'id':'__VIEWSTATE'})['value']
+
+				data = {'__VIEWSTATE':viewstate,
+					'ctl00$MainContent$txtLastName':self._name,
+					'ctl00$MainContent$btnSearch':'Search',}
+				req = self.framework.request(
+						url=self.sanctionsearch,
+						method='POST',
+						data=data,
+						headers=headers)
+
+			except:
+				self.framework.error('[SANCTIONSEARCH] ConnectionError')
+				self.framework.error('Sanctionsearch is missed!')
+
+			soup = bs(req.text,'html.parser')
+			table = soup.find('table', {'id':'gvSearchResults'})
+
+			if table is not None:
+				self._rows = table.find_all('tr')
+			else:
+				self._rows = None
+
+			if self._rows is not None:
+				for count, row in enumerate(self._rows):
+					if count>=self._max:
+						break
+
+					name = row.find('a').text
+					address = row.find_all('td')[1].text
+					link = row.find('a')['href']
+
+					self._data.append({
+						'name': name,
+						'address': address,
+						'link': self.sanctionsearch+link
+						})
+
+
+		def id_crawl(self):
+			self._data = {}
+			url = self.sanctionsearch + f'Details.aspx?id={self._id}'
+			req = self.framework.request(url=url)
+			soup = bs(req.text,'html.parser')
+
+			# DETAILS
+			details_table = soup.find('table',
```

Comment:
space error

Reply:
parameter '=' should never be surrounded by spaces

Reply:
If you have a good source for it, we can change it comprehensively.

Reply:
sure give me a sec

Reply:
https://www.flake8rules.com/rules/E251.html

Reply:
This is for function parameters, not the variables.

Reply:
> parameter '=' should never be surrounded by spaces

yes i was talking about function parameters

Reply:
my bad I thought you were talking about that